api/controllers/FlairController.js

```diff
@@ -50,17 +50,51 @@ module.exports = {
       }
       var user = await User.findOne(app.user);
       var shortened = app.sub === 'pokemontrades' ? 'ptrades' : 'svex';
-      var relevant_flair = Flairs.makeNewCSSClass(_.get(user, 'flair.' + shortened + '.flair_css_class') || '', app.flair, app.sub);
-      user.flair[shortened].flair_css_class = relevant_flair;
-      await Reddit.setUserFlair(req.user.redToken, user.name, relevant_flair, user.flair[shortened].flair_text, app.sub);
+      var css_flair = Flairs.makeNewCSSClass(_.get(user, 'flair.' + shortened + '.flair_css_class') || '', app.flair, app.sub);
+      user.flair[shortened].flair_css_class = css_flair;
+        
+      // If a ptrades app, update flair_text emoji
+        
+      // Mappings from css_flair to emoji names
+      var flair_mappings = {
+        "default" : ":0:",
+        "gen2" : ":2:",
+        "pokeball" : ":10:",
```

Comment:
This could cause some problems because the flair text will get longer. If someone's flair text is 64 characters long already, then it won't be possible to approve their flair app because giving them the new flair would put them over the limit.

Reply:
What do you think the best solution to this would be? Should I just check the length of the flair text before sending the Reddit request and throw an error?

Reply:
I'd check the length of the flair text before sending the request, and if it's too long, we should just set the flair text to only the emoji (or emoji + FC) and let the user go back to FHQ to fix their text if they want to. It would be a hassle to deal with errors if you're just trying to get through the flair apps.

(fwiw, this is the approach I'm taking with the script to add new reddit flair to all existing users)

Reply:
Okay done!

Reply:
If I could offer an alternative suggestion, I would consider adding a limit of 55 characters for their flair text so there will always be 6+3 characters available for left and right side flairs when it's set via FlairHQ going forth.

I also think a script for adding emoji to existing flair should ideally look for extraneous elements such as extra IGNs to discard to get flair text under this limit before resorting to removing flair entirely, but even having FlairHQ enforce this limit when they're updating their flair should prevent most cases of flair text suddenly being lost due to a flair update.

Reply:
I believe NAA's comment was regarding if they currently have a flair app in that would potentially get longer, such as if they're applying from 90 trades -> 100 trades etc... (this shouldn't happen if we deploy these changes right after Raia runs her script). 

We could and probably should do a limit of 55 characters for their flair text when they're setting it to prevent these situations though, that's a good point. I'll put that in tonight!